nemo/core/optim/lr_scheduler.py

```diff
@@ -640,6 +640,119 @@ def _get_lr(self, step):
         return new_lrs
 
 
+class LinearWarmupHoldDecayPolicy(_LRScheduler):
+    """
+    Has a linear warmup phase from 0 to the optimizer's learning rate, then a
+    constant phase, followed by a linear decay to the `min_lr`, after which
+    the learning rate is held constant.
+
+    E.g.,
+    lr
+    |        _____________  <- optimizer.lr
+    |       /             \
+    |      /               \
+    |     /                 \
+    |    /                   \
+    |   /                     \
+    |  /                       \_______________ <- min_lr
+    | /
+    |/                          |<- max_steps
+    |------------------------------------------- steps
+         |<- warmup_steps |<- constant_steps
+
+    Args:
+        optimizer: Optimizer for the scheduler. In the warmup stage, the
+            scheduler linearly increases the learning rate from 0 to the rate
+            specified by the optimizer.
+        warmup_steps: Number of training steps in warmup stage
+        warmup_ratio: Ratio of warmup steps to total steps. Mutually exclusive
+            with `warmup_steps`
+        max_steps: Total number of steps while training or `None` for
+            infinite training
+        min_lr: Minimum lr to hold the learning rate after decay at.
+        constant_steps: Number of steps to keep lr constant at.
+        constant_ratio: Ratio of steps to keep lr constant. Mutually exclusive
+            with `constant_steps`.
+    """
+
+    def __init__(
+        self,
+        optimizer,
+        *,
+        warmup_steps=None,
+        warmup_ratio=None,
+        constant_steps=None,
+        constant_ratio=None,
+        max_steps=None,
+        min_lr=0.0,
+        last_epoch=-1,
+    ):
+        assert not (
+            warmup_steps is not None and warmup_ratio is not None
+        ), "Either use particular number of step or ratio"
+        assert not (
+            constant_steps is not None and constant_ratio is not None
+        ), "Either use constant_steps or constant_ratio"
+        assert warmup_ratio is None or max_steps is not None, "If there is a " "ratio, there should be a total steps"
```

Comment:
Could you use f-strings to display the values of the offending variables in the error message?